docs/signature.md

```diff
@@ -0,0 +1,93 @@
+# Image Signature Specification
+
+**Version 0.1**
+
+## Introduction
+
+This document defines a detached container image signature object and signing methods.
+
+## Signature Format
+
+```js
+{
+    "critical": {/* required fields */
```

Comment:
> … but those are very much anti-features for signatures…

Agreed.  I was [pointing out][1] that you can also strictly-type signatures at the signature-schema level instead of at the property level.  And that's even *less* liberal in what you accept (“Is that a v1.1 signature?  Sorry, I only accept v1.0 signatures.”), although the difference there is only for optional fields.  And while I'm generally not a fan of peek-inside typing (lots of back and forth in opencontainers/image-spec#411), I think we probably *do* want to have something like:

> All signature blobs MUST start with the ASCII bytes representing their media type and a trailing newline.

So you can't trick verifiers by getting them to interpret the signature blob as a media type other than the one which the signer intended.

[1]: https://github.com/containers/image/pull/59#discussion_r86185469